File: wms-be/src/main/java/com/restapi/wmsservice/service/ItemService.java
package com.restapi.wmsservice.service;

import com.restapi.wmsservice.dto.request.ItemRequest;
import com.restapi.wmsservice.dto.response.ItemResponse;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface ItemService {
    ItemResponse create(ItemRequest request);
    List<ItemResponse> getAll();
    ItemResponse getById(Long id);
    ItemResponse update(Long id, ItemRequest request);
    ItemResponse updateImage(Long id, MultipartFile file);
    ItemResponse deleteImage(Long id);
    void delete(Long id);
}

File: wms-fe/src/services/itemService.js
import axiosClient from '../api/axiosClient';

const itemService = {
  getAll: (params) => {
    return axiosClient.get('/items', { params });
  },
  
  getById: (id) => {
    return axiosClient.get(`/items/${id}`);
  },
  
  create: (data) => {
    return axiosClient.post('/items', data);
  },
  
  update: (id, data) => {
    return axiosClient.put(`/items/${id}`, data);
  },

  uploadImage: (id, file) => {
    const formData = new FormData();
    formData.append('file', file);
    return axiosClient.put(`/items/${id}/image`, formData, {
      headers: { 'Content-Type': 'multipart/form-data' },
    });
  },

  deleteImage: (id) => axiosClient.delete(`/items/${id}/image`),
  
  delete: (id) => {
    return axiosClient.delete(`/items/${id}`);
  },
};

export default itemService;
